src/app/tests/suites/commands/interaction_model/InteractionModel.h

```diff
@@ -139,6 +135,8 @@ class InteractionModelReports
                          const chip::Optional<chip::EventNumber> & eventNumber               = chip::NullOptional,
                          const chip::Optional<bool> & keepSubscriptions                      = chip::NullOptional);
 
+    void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams);
+
     void Shutdown() { mReadClients.clear(); }
```

Comment:
This is going to invoke `~ReadClient`, which I don't think automatically free the paths allocated...

Reply:
Hmm, good catch.

But what is the public API for "I am done for this subscription"?   I don't see one....

Reply:
As in, we should ideally call `Close(CHIP_NO_ERROR)` here, but it's not public.

Reply:
So, I think the destructor should handle that, since our general policy is that you should be able to destroy the client objects at any time and all the good things happen automatically.. (i.e you shouldn't need to go through a `Close` or `Abort` style API).

Needs a bug filed I think if you agree with that.

Reply:
And if we go with that, we can leave this un-addressed for now (but leak some memory), and have it be addressed as a follow-up quickly.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/21658